File: Codigos estudiantes por lenguaje/Java/Daniel Cangrejo/Ordenamientos/Pigeon_Hole_Sort.java
import java.util.Arrays;

//PIGEON_HOLE_SORT
public class Ordenamientos2_7 {
	
	static void pigeonSort(int arr[], int n) {
		int min = arr[0];
		int max = arr[0];
		int range, i, j, index;
		for (int a = 0; a < n; a++) {
			if(arr[a] > max) {
				max = arr[a];
			}
			if(arr[a] < min) {
				min = arr[a];
			}			
		}
		range = max - min + 1;
		int [] phole = new int [range];
		Arrays.fill(phole, 0);
		for (i = 0; i < n; i++) {
			phole[arr[i]-min]++;
		}
		index = 0;
		for (j = 0; j < range; j++) {
			while (phole[j]-- > 0) {
				arr[index++] = j + min;
			}
		}
	}

	public static void main(String[] args) {
		int arr[] = {8, 3, 2, 7, 4, 6, 8};
		int n = arr.length;
		pigeonSort(arr, n);
		for (int i : arr) {
			System.out.print(i + " ");
		}

	}

}

File: Codigos estudiantes por lenguaje/PY/Daniel Cangrejo/Ordenamientos/Pigeon_Hole_Sort.py
from sys import stdin, stdout
rl = stdin.readline
wr = stdout.write


def pigeonHoleSort(arr, n):

    min_ = arr[0]
    max_ = arr[0]

    for a in range(n):
        if arr[a] > max_:
            max_ = arr[a]
        if arr[a] < min_:
            min_ = arr[a]

    rango = max_ - min_ + 1
    phole = [0] * rango

    for i in range(n):
        phole[arr[i] - min_] += 1

    index = 0
    for j in range(rango):
        while phole[j] > 0:
            phole[j] -= 1
            arr[index] = j + min_
            index += 1


arr = list(map(int, rl().split()))
pigeonHoleSort(arr, len(arr))
print(*arr)
